chain/client/src/stateless_validation/partial_witness/partial_witness_tracker.rs

```diff
@@ -188,19 +170,43 @@ impl PartialEncodedStateWitnessTracker {
     ) -> Result<(), Error> {
         tracing::debug!(target: "client", ?partial_witness, "store_partial_encoded_state_witness");
 
-        self.maybe_insert_new_entry_in_parts_cache(&partial_witness)?;
-
         let key = partial_witness.chunk_production_key();
-        let entry = self.parts_cache.get_mut(&key).unwrap();
+        if self.processed_witnesses.contains(&key) {
+            return Err(Error::InvalidPartialChunkStateWitness(format!(
```

Comment:
as far as I remember, we process the witness when the enough number of parts arrive. thus, if some parts arrive late (even if not malformed) will they cause an error here? if so, should not that be a silent debug message and skip the rest of the code?

Reply:
good point, done in 06d663f00a307978706a20de5541fcf53911b2d3